### PR TITLE
[WIP] Upgrade logging module 

### PR DESCRIPTION
**Pull request by bartlomieju**
This is just prototyping.

I got logging to file working, still iterating on the API. Comments welcome.

### CODE REVIEW

**Comment by bartlomieju** [comment]
It's usable for simple cases now, parameter interpolation `log.debug("magic number: {}", 42);` is not yet implemented.

File: logging/handlers.ts [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+  constructor(levelName: string) {
+    this.level = getLevelByName(levelName);
+    this.levelName = levelName;
+  }
+
+  handle(logRecord: LogRecord) {
+    if (this.level > logRecord.level) return;
+
+    // TODO: implement formatter
```
Comment (ry) [inline logging/handlers.ts]: Maybe use deno.inspect ?

File: logging/test.ts [hunk 2]
```diff
@@ -1,53 +1,84 @@
+import { remove, open, readAll } from "deno";
 import { assertEqual, test } from "https://deno.land/x/testing/testing.ts";
 
 import * as logging from "index.ts";
```
Comment (ry) [inline logging/test.ts]: Rename to “log”

File: logging/handlers.ts (hunk 1 above)
Comment (bartlomieju) [reply logging/handlers.ts]: I had I mind something like: 
```
{levelName} [{datetime}] {msg}
``` 
which gives: 
```
DEBUG [2018-12-30T20:11:53] Some message
``` 

With some sane default.

Comment (ry) [reply logging/handlers.ts]: I see. Another thought: can you use JS template literals? https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals

Comment (bartlomieju) [reply logging/handlers.ts]: I am not sure I understand. If we use template strings I won't be possible to format arguments. Can you give an example what you had in mind?

Comment (ry) [reply logging/handlers.ts]: Hmm. I don't have something specific in mind now.  JS has some nice functionality now for doing formatting and it would be good to take advantage of that if possible. I'll get back to you on that...

File: logging/test.ts [hunk 3]
```diff
@@ -17,21 +17,21 @@ class TestHandler extends logging.handlers.BaseHandler {
...
 }
 
-test(function testDefaultLoggingMethods() {
-  logging.debug("Foobar");
-  logging.info("Foobar");
-  logging.warning("Foobar");
-  logging.error("Foobar");
-  logging.critical("Foobar");
+test(function testDefaultlogMethods() {
+  log.debug("Foobar");
```
Comment (ry) [inline logging/test.ts]: Cool - shall we also rename the directory too now? Or do you want to do that in another PR?

Comment (bartlomieju) [reply logging/test.ts]: Let's finish stuff in this PR and rename directory in next one.

File: logging/handlers.ts (hunk 1 above)
Comment (bartlomieju) [reply logging/handlers.ts]: Sure, I'm curious
